api/v1alpha3/types.go

```diff
@@ -138,6 +138,12 @@ type VirtualMachineCloneSpec struct {
 	// Defaults to empty map
 	// +optional
 	CustomVMXKeys map[string]string `json:"customVMXKeys,omitempty"`
+	// ProvisionerType is the name of provisioner type to use. Can be either cloud-init or
+	// ignition. Defaults to cloud-init.
+	// +optional
+	// +kubebuilder:validation:Enum:=cloud-init;ignition
+	// +kubebuilder:default=cloud-init
+	ProvisionerType string `json:"provisionerType,omitempty"`
```

Comment:
this should be defaulted at the `VSphereMachine` level (default should be `cloud-init`)

Reply:
@yastij Do you have any concerns about OpenAPI defaulting vs defaulting inside the machine controller? Defaulting the value using scheme might make it more obvious to the user, WDYT?